src/item.h

```diff
@@ -1184,56 +1183,99 @@ class item : public visitable
+        * damage_type.
+        */
+        bool damage_type_can_damage_items(damage_type dmg_type) const;
+
         /**
          * Resistance against different damage types (@ref damage_type).
          * Larger values means more resistance are thereby better, but there is no absolute value to
          * compare them to. The values can be interpreted as chance (@ref one_in) of damaging the item
          * when exposed to the type of damage.
+         * @param dmg_type The type of incoming damage
+         * @param to_self If this is true, it returns item's own resistance, not one it gives to wearer.
+         * @param bodypart_target The bodypart_id or sub_bodypart_id of the target bodypart.
+         * @param resist_value A supplied roll against coverage for physical attacks. A value of -1 causes rolls
+         *                     against each item material individually. For environmental type damage such as
+         *                     fire and acid, this value can be used to allow hypothetical calculations for
+         *                     environmental protection items such as gas masks.
+         */
+        /*@{*/
+
+        template<typename bodypart_target = bodypart_id>
+        typename std::enable_if_t < std::is_same<bodypart_target, bodypart_id>::value ||
+        std::is_same<bodypart_target, sub_bodypart_id>::value, float >
+        resist( const damage_type dmg_type, bool to_self = false,
+                const bodypart_target &bp = bodypart_target(),
+                int resist_value = 0 ) const {
+            if( is_null() ) {
+                return 0.0f;
+            }
+
+            if( damage_type_none( dmg_type ) ) {
+                return 0.0f;
+            }
+
+            if(damage_type_invalid( dmg_type ) ) {
+                debugmsg( "Invalid damage type: %d", dmg_type );
+                return 0.0f;
+            }
+
+            // Implicit item damage immunity for damage types listed prior to bash
+            // Acid/fire immunity would be handled in _environmental_resist, but there are more
+            // dmg types that do not affect items such as PURE/COLD etc..
+            if( to_self && !damage_type_can_damage_items( dmg_type ) ) {
+                return std::numeric_limits<float>::max();
+            }
+
+            const bool bp_null = bp == bodypart_target();
+            const std::vector<const part_material *> &armor_mats = armor_made_of( bp );
+            const float avg_thickness = bp_null ? get_thickness() : get_thickness( bp );
+            return _resist( dmg_type, to_self, resist_value, bp_null, armor_mats, avg_thickness );
+        }
+
+    private:
+        float _resist( damage_type dmg_type, bool to_self = false, int resist_value = 0,
+                       bool bp_null = true,
+                       const std::vector<const part_material *> &armor_mats = std::vector<const part_material *>(),
```

Comment:
```suggestion
                       const std::vector<const part_material *> &armor_mats = {},
```
same thing for `_environmental_resist` below